### PR TITLE
Resolve app/code/Magento/Customer/view/frontend/templates/address/grid.phtml call non-exist function getBackUrl (issue 24058)

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

Resolve app/code/Magento/Customer/view/frontend/templates/address/grid.phtml call non-exist function getBackUrl (issue 24058)

Solution: `getBackUrl `is missing in `Magento\Customer\Block\Address\Grid.php` (@api class). So we should use `getUrl('customer/account')` instead of `getBackUrl()` in `app/code/Magento/Customer/view/frontend/templates/address/grid.phtml `

### Fixed Issues (if relevant)

1. magento/magento2#24058: Magento\Customer\Block\Address\Grid.php getBackUrl()

### Manual testing scenarios (*)

- Customer account exists and has at least a billing or shipping address
- The theme used is "Blank" instead of "Luma"

1. Go to frontend, Login to account
2. Click **Manage Addresses** from a customer account
image
3. Click "Back" at the right bottom 
image

**Expected**: Back button in form should go to the **customer account**

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-5572](https://jira.corp.magento.com/browse/ENGCOM-5572) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed 

**Before:**
before

**After:**
after

File: app/code/Magento/Customer/view/frontend/templates/address/grid.phtml [hunk 1]
```diff
@@ -63,7 +63,7 @@ $customerAddressView = $block->getData('customer_address');
         <button type="button" role="add-address" title="<?= $block->escapeHtmlAttr(__('Add New Address')) ?>" class="action primary add"><span><?= $block->escapeHtml(__('Add New Address')) ?></span></button>
     </div>
     <div class="secondary">
-        <a class="action back" href="<?= $block->escapeUrl($block->getBackUrl()) ?>"><span><?= $block->escapeHtml(__('Back')) ?></span></a>
+        <a class="action back" href="<?= $block->escapeUrl($block->getUrl('customer/account')) ?>"><span><?= $block->escapeHtml(__('Back')) ?></span></a>
```
Comment (nikolasok) [inline app/code/Magento/Customer/view/frontend/templates/address/grid.phtml:66]: Why is it redirected to this page? The back button means real back, but not constantly going to the same page.

**Review by nikolasok — COMMENTED** [review]
The real solution:
Add to module-customer/Block/Address/Grid.php this method

                
    public function getBackUrl()
    {
        // the RefererUrl must be set in appropriate controller
        if ($this->getRefererUrl()) {
            return $this->getRefererUrl();
        }
        return $this->getUrl('customer/address/');
    }


And change controller module-customer/Controller/Address/Index.php:78
from
`$block = $resultPage->getLayout()->getBlock('address_book');`
to
`$block = $resultPage->getLayout()->getBlock('address_grid');`

**Pull request merged**